### ISSUE TITLE
Bug: Strings contains UTF8 break regpack

### ISSUE THREAD

**Issue by neophob** [Siorki/RegPack#64]
Hi

I have an UTF8 character in a string: `c.fillText('✪', 5, 165);` -> using this line breaks REgPack. When i comment out this line regpack works

Error:
```
/Users/user/_code/github/js1k17/node_modules/regpack/regPack.js:126
		return encodeURI(inString).replace(/%../g,'i').length;
		       ^

URIError: URI malformed
    at encodeURI (native)
    at Object.getByteLength (/Users/user/_code/github/js1k17/node_modules/regpack/regPack.js:126:10)
    at Object.findRedundancies (/Users/user/_code/github/js1k17/node_modules/regpack/regPack.js:179:12)
    at Object.runPacker (/Users/user/_code/github/js1k17/node_modules/regpack/regPack.js:102:22)
    at cmdRegPack (/Users/user/_code/github/js1k17/node_modules/regpack/regPack.js:27:25)
    at Object.<anonymous> (/Users/user/_code/github/js1k17/node_modules/regpack/bin/regpack:8:14)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
```

**Comment by Siorki** [comment]
I managed to get that piece of code through both browser and node versions of RegPack without getting the same error. I added the following test, which runs fine : 
```javascript
function testEncodeURI() {
	var input = fs.readFileSync("../TestCases/gitHub#64-URIError.js", { encoding:"utf8"});
	var options = {
			withMath : false,
			hash2DContext : false,
			hashWebGLContext : false,
			hashAudioContext : false,
			contextVariableName : false,
			contextType : parseInt(0),
			reassignVars : false,
			varsNotReassigned : [],
			crushGainFactor : parseFloat(1),
			crushLengthFactor : parseFloat(0),
			crushCopiesFactor : parseFloat(0),
			crushTiebreakerFactor : parseInt(1),
			wrapInSetInterval : false,
			timeVariableName : ""
		};
	var result = RegPack.packer.runPacker(input, options);
	
	// Expected result : internal check successful, 
	// and the unicode characters are excluded from the token range
	assert.notEqual(result[0].result[2][1].indexOf("uffff]"), -1);
	assert.notEqual(result[0].result[2][2].indexOf("Final check : passed"), -1);
}
```
The file `gitHub#64-URIError.js` contains `c.fillText('✪', 5, 165)`

Can you please tell me : 
 - the OS and version of Node.js you are running
 - if you can successfully run all unit tests `node allTests` in directory `tests/`
 - if your code has other instances of that Unicode character ✪ (I suspect something similar to issue #50)
 - if RegPack runs correctly after replacing ✪ with any ASCII character in that line of code

**Comment by neophob** [comment]
Thanks @Siorki for the quick answer. You're right, the test case works. If you update the testcase to contain the string `c.fillText('✪🅼🅼🅼', 5, 165)` the test will fail. 

I'm using node 6.9.1 here.

**Comment by Siorki** [comment]
That second example fails on the same bug as #50 : the crusher defines a substring starting with the second (low) surrogate character (DC00-DFFF). Unfortunately, such a string is malformed and causes the exception.

Can you confirm you have other astral characters (code point >=0x10000) in your code ?

**Comment by neophob** [comment]
My complete string i use is '🆃🅷🅸🆂 🅸🆂 🅽🅾🆃 🅰 🆃🅴🆂🆃'

**Comment by Siorki** [comment]
Confirmed duplicate of #50 - fixed in v5.0.0
